src/sentry/migrations/0702_alert_rule_project_backfill_migration_2.py

```diff
@@ -0,0 +1,106 @@
+# Generated by Django 5.0.3 on 2024-04-19 17:53
+
+
+import logging
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapper
+
+logger = logging.getLogger(__name__)
+
+
+def _backfill_alert_rule_projects(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    QuerySubscriptions = apps.get_model("sentry", "QuerySubscription")
+    AlertRuleProjects = apps.get_model("sentry", "AlertRuleProjects")
+
+    # use RangeQuerySetWrapper to avoid loading all subscriptions into memory
+    for subscription in RangeQuerySetWrapper(QuerySubscriptions.objects.all()):
+
+        snuba_query = subscription.snuba_query
```

Comment:
You could `select_related('snuba_query')` on your original query to save an N+1 which could help your migration run faster.

Reply:
@markstory hows this look?
`QuerySubscriptions.objects.all().prefetch_related("snuba_query__alertrule")`